vars/deploy.groovy

```diff
@@ -0,0 +1,159 @@
+                    if(enableZeroDowntimeDeployment) {
+                        deploymentType = DeploymentType.CF_BLUE_GREEN.toString()
+                    }
+                    else {
+                        deploymentType = DeploymentType.selectFor(
+                            CloudPlatform.CLOUD_FOUNDRY,
+                            config.isProduction.asBoolean()
+                        ).toString()
+                    }
+
+                    def deployTool =
+                        (script.commonPipelineEnvironment.configuration.isMta) ? 'mtaDeployPlugin' : 'cf_native'
+
+                    cloudFoundryDeploy(
+                        script: script,
+                        juStabUtils: utils,
+                        jenkinsUtilsStub: jenkinsUtils,
+                        deployType: deploymentType,
+                        cloudFoundry: target,
+                        mtaPath: script.commonPipelineEnvironment.mtarFilePath,
+                        deployTool: deployTool
+                    )
+
+                    utils.stash(stageName)
+                }
+                deployments["Deployment ${index > 1 ? index : ''}"] = {
+                    if (env.POD_NAME) {
+                        dockerExecuteOnKubernetes(script: script, containerMap: ContainerMap.instance.getMap().get(stageName) ?: [:]) {
+                            deployment.run()
+                        }
+                    } else {
+                        node(env.NODE_NAME) {
+                            deployment.run()
+                        }
+                    }
+                }
+                index++
+            }
+            utils.runClosures(deployments)
+        }
+        if (config.neoTargets) {
+            def source
+            if (config.source) {
+                source = config.source
+            } else {
+                def pom = readMavenPom file: 'application/pom.xml'
+                source = "application/target/${pom.getArtifactId()}.${pom.getPackaging()}"
+            }
+
+            for (int i = 0; i < config.neoTargets.size(); i++) {
+                def target = config.neoTargets[i]
+
+                Closure deployment = {
+                    utils.unstash(stageName)
+
+                    DeploymentType deploymentType
+                    if(enableZeroDowntimeDeployment) {
+                        deploymentType = DeploymentType.NEO_ROLLING_UPDATE
+                    }
+                    else {
+                        deploymentType = DeploymentType.selectFor(CloudPlatform.NEO, config.isProduction.asBoolean())
+                    }
+
+                    neoDeploy (
+                        script: script,
+                        warAction: deploymentType.toString(),
+                        source: source,
+                        neo: target
+                    )
+
+                    utils.stash(stageName)
+                }
+                deployments["Deployment ${index > 1 ? index : ''}"] = {

```

Comment:
This here repeats line 90 ff. --> I would like to propose to condence the common code in order to have it only once.